pkg/manager/member/dm_worker_member_manager.go

```diff
@@ -259,6 +266,19 @@ func (wmm *workerMemberManager) syncDMClusterStatus(dc *v1alpha1.DMCluster, set
 		}
 
 		workerStatus[name] = status
+
+		// offline the workers that already been scaled-in
+		if status.Stage == "offline" {
+			ordinal, err := util.GetOrdinalFromPodName(worker.Name)
+			if err != nil {
+				klog.Errorf("invalid worker name %s, can't offline this worker automatically, err: %s", worker.Name, err)
+			} else if ordinal >= dc.WorkerStsDesiredReplicas() {
```

Comment:
Please refer to https://github.com/pingcap/tidb-operator/blob/master/pkg/manager/member/tikv_failover.go#L38-L45 for how to check if an ordinal is not desired or not in case of advanced statefulset is enabled.